src/core/common/debug.hpp

```diff
@@ -73,21 +73,23 @@
 
 #else
 
-#ifdef assert
-#undef assert
-#endif
-
-#define assert(cond)  \
-    do                \
-    {                 \
-        if (!(cond))  \
-        {             \
-            while (1) \
-            {         \
-            }         \
-        }             \
+#define OT_ASSERT(cond) \
+    do                  \
+    {                   \
+        if (!(cond))    \
+        {               \
+            while (1)   \
+            {           \
+            }           \
+        }               \
     } while (0)
 
-#endif
+#endif // OPENTHREAD_CONFIG_PLATFORM_ASSERT_MANAGEMENT
+
+#else
+
+#define OT_ASSERT(cond)
```

Comment:
```suggestion
#define OT_ASSERT(cond) ((void)0)
```
otherwise some lint program may complain about empty statements.

Reply:
We actually do use a lot of empty macros for OT logging (`otLogDebg`, etc.) already.
I would suggest we keep this as is and use empty macro here as well. We can change them all if/when there is an issue with using empty macros.